codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperSyncWorker.res

```diff
@@ -110,19 +110,36 @@ module Make = (
     }
   }
 
+  let wildcardTopicSelection = T.contracts->Belt.Array.flatMap(contract => {
+    contract.events->Belt.Array.keepMap(event => {
+      let module(Event) = event
+      let {isWildcard, topicSelections} =
+        Event.handlerRegister->Types.HandlerTypes.Register.getEventOptions
+      isWildcard ? Some(LogSelection.make(~addresses=[], ~topicSelections)) : None
+    })
+  })
+
   let getLogSelectionOrThrow = (~contractAddressMapping): array<LogSelection.t> => {
-    T.contracts->Belt.Array.keepMap((contract): option<LogSelection.t> => {
+    T.contracts
+    ->Belt.Array.keepMap((contract): option<LogSelection.t> => {
       switch contractAddressMapping->ContractAddressingMap.getAddressesFromContractName(
         ~contractName=contract.name,
       ) {
       | [] => None
       | addresses =>
-        let topicSelection =
-          LogSelection.makeTopicSelection(~topic0=contract.sighashes)->Utils.unwrapResultExn
-
-        Some(LogSelection.make(~addresses, ~topicSelections=[topicSelection]))
+        switch contract.events->Belt.Array.flatMap(event => {
+          let module(Event) = event
+          let {isWildcard, topicSelections} =
+            Event.handlerRegister->Types.HandlerTypes.Register.getEventOptions
+
+          isWildcard ? [] : topicSelections
+        }) {
```

Comment:
A nitpick, but we can also precalculate the topicSelections on the ChainWorker creation and only join them with addresses here

Reply:
Yes but we can only apply those based on the addresses in the contractAddressMapping. So we could create a premade lookup table essentially of contract->topic selection and on each contract in the contractAddressMapping make a LogSelection with the topics and addresses. That is not far off from what's happenening here but we could probably improve it slightly 👍🏼